### PR TITLE
Bugfix: On Skip discard right nozzle

### PR DESCRIPTION
**Pull request by mgrl**
# Description
until today if skipping a part due to an error (vac/aligning failed) always nozzle 1 is cleared per discarding a part into the discard box. this patch fixes this an takes the right nozzle into account.

# Justification
it's very annoying to have the wrong nozzle cleared and investigate which part was assembled, which not and remember which parts to fix in next run...

# Instructions for Use
click "skip" if an error occured. now the correct nozzle is cleared and the assembly for the rest goes on.

# Implementation Details
1. How did you test the change? Be descriptive. Untested code will not be accepted.
tested on my machine with 2 nozzles on one head.
I tested 2 cases:
1: removed nozzle tip from nozzle 1. pickup from feeder failed of course, vacuum sense check failed and it gave me the option to skip/retry, etc. -> clicked skip -> nozzle 1 went to discard location, discarded and went with nozzle 2 to the feeder and continues as usual.
2:  removed nozzle tip from nozzle 2. pickup from feeder failed of course, vacuum sense check failed and it gave me the option to skip/retry, etc. -> clicked skip -> nozzle 2 went to discard location, discarded and went with nozzle 1 to the feeder and continiues as normal.

2. Did you follow the [coding style](https://github.com/openpnp/openpnp/wiki/Developers-Guide#coding-style)?
yes
3. If you made changes in the `org.openpnp.spi` or `org.openpnp.model` packages you will need to add additional justification for these changes. Changes to these packages require extensive review and testing.
no changes
4. Be sure to run `mvn test` before submitting the Pull Request. If the tests do not pass the Pull Request will not be accepted.
done.

### CODE REVIEW

File: src/main/java/org/openpnp/machine/reference/ReferencePnpJobProcessor.java [hunk 1]
```diff
@@ -816,12 +816,35 @@ protected void doReset() throws Exception {
...
+            for (PlannedPlacement plannedPlacement : plannedPlacements) {
+                JobPlacement jobPlacement = plannedPlacement.jobPlacement;
+
+                if (plannedPlacement.stepComplete) {
+                    // go over placements having the current step already completed
+                    continue;
+                }
+
+                // remove the placement to be skipped from list
+                plannedPlacements.remove(plannedPlacements.indexOf(plannedPlacement));
```
Comment (vonnieda) [inline src/main/java/org/openpnp/machine/reference/ReferencePnpJobProcessor.java]: This is likely to cause a ConcurrentModificationException. You can't modify a collection that you are iterating unless you use an Iterator and the Iterator.remove() method. See https://stackoverflow.com/questions/18448671/how-to-avoid-concurrentmodificationexception-while-removing-elements-from-arr. Other that that, this change looks solid!

**Comment by mgrl** [comment]
updated according to review. indeed the ConcurrentModificationException was not raised since only one element was called and loop left afterwards. but this way it's better obviously.

**Comment by netzmark** [comment]
Hi Michael, Jason
Tested it today. Seems be working just properly!

**Comment by mgrl** [comment]
yay! :)

**Pull request merged**